src/components/experimental/Dialog/Dialog.tsx

```diff
@@ -0,0 +1,69 @@
+import React, { ReactElement, ReactNode } from 'react';
+import { Heading } from 'react-aria-components';
+import styled from 'styled-components';
+import { Text, textStyles } from '../Text/Text';
+import { Modal } from '../Modal/Modal';
+import { Backdrop, BackdropProps } from '../Backdrop/Backdrop';
+import { getSemanticValue } from '../../../essentials/experimental';
+
+const Card = styled.div`
+    display: grid;
+    gap: 0.5rem;
+`;
+
+const ButtonsWrapper = styled.div`
+    padding-top: 1.5rem;
+    display: flex;
+    flex-direction: row;
+    justify-content: flex-end;
+    gap: 1rem;
+`;
+
+const StyledModal = styled(Modal)`
+    width: 30rem;
+`;
+
+const HeadlineText = styled(Heading)`
+    margin: 0;
+    ${textStyles.variants.headline}
+`;
+
+const SubtitleText = styled(Text)`
+    color: ${getSemanticValue('on-surface-variant')};
+`;
+
+interface DialogProps extends Omit<BackdropProps, 'isDismissable' | 'isKeyboardDismissDisabled'> {
+    role?: 'dialog' | 'alertdialog';
+    headline: ReactNode;
+    subtitle: ReactNode;
+    dismissButton: ReactNode;
+    actionButton: ReactNode;
+}
+
+const Dialog = ({
+    role = 'dialog',
+    headline,
+    subtitle,
+    dismissButton,
+    actionButton,
+    ...props
+}: DialogProps): ReactElement => (
+    <Backdrop {...props} isDismissable={false} isKeyboardDismissDisabled>
+        <StyledModal role={role}>
+            <Card>
+                <HeadlineText slot="title">{headline}</HeadlineText>
+
+                <SubtitleText as="p" variant="body1">
```

Comment:
Maybe not within the scope of the PR, but can we name the variants with a more semantic name? `body1`, `body2` etc. don't help me understand what the variant does

Reply:
It's named like the designers name it and what they use in the designs